### PR TITLE
SNOW-896752 - Add node20 on Ubuntu to github action tests

### PR DESCRIPTION
**Pull request by sfc-gh-pmotacki**
### Description
Please explain the changes you made here.

### Checklist
- [ ] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [ ] Create tests which fail without the change (if possible)
- [ ] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -129,3 +129,32 @@ jobs:
...
               run: ./ci/test.sh
+    test-ubuntu:
+      needs: build
+      name: Tests on Ubuntu
+      runs-on: ubuntu-latest
+      strategy:
+        fail-fast: false
+        matrix:
+          cloud: [ 'AWS', 'AZURE', 'GCP' ]
+          nodeVersion: ['18.x', '20.x']
```
Comment (sfc-gh-dprzybysz) [inline .github/workflows/build-test.yml:140]: don't we want for now to also add 14.x and 16.x?

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -129,3 +129,32 @@ jobs:
...
+        - uses: actions/checkout@v1
+        - name: Download Build Artifacts
+          uses: actions/download-artifact@v1
+          with:
+            name: artifacts
+        - name: Use Node.js
+          uses: actions/setup-node@v3
+          with:
+            node-version: ${{ matrix.node-version }}
+        - name: Install dependencies
```
Comment (sfc-gh-dprzybysz) [inline .github/workflows/build-test.yml:151]: shouldn't installing dependencies be a part of the test script?

File: ci/test_ubuntu.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Test NodeJS for Ubuntu
+#
+set -o pipefail
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source $THIS_DIR/_init.sh
+source $THIS_DIR/scripts/set_git_info.sh
+
+export WORKSPACE=$GITHUB_WORKSPACE
+export SOURCE_ROOT=$GITHUB_WORKSPACE
+export SHOULD_GENERATE_COVERAGE_REPORT=1
```
Comment (sfc-gh-dprzybysz) [inline ci/test_ubuntu.sh:12]: when it's 1 - should we gathered results and send them to codecov?

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (sfc-gh-pmotacki) [reply .github/workflows/build-test.yml:140]: We have tested all versions on linux - centos7. It me opinion it is enough. It is a matrix and every next node version run 3 new test processes (for every cloud). We have plans to drop support for versions <1 8

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (sfc-gh-pmotacki) [reply .github/workflows/build-test.yml:151]: Script (test_ubuntu.sh) responsibility is testing. We have building as separated step.

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -129,3 +129,32 @@ jobs:
...
+      needs: build
+      name: Tests on Ubuntu
+      runs-on: ubuntu-latest
+      strategy:
+        fail-fast: false
+        matrix:
+          cloud: [ 'AWS', 'AZURE', 'GCP' ]
+          nodeVersion: ['18.x', '20.x']
+      steps:
+        - uses: actions/checkout@v1
```
Comment (sfc-gh-pbulawa) [inline .github/workflows/build-test.yml:142]: Could you use v3 version of checkout?

File: .github/workflows/build-test.yml [hunk 5]
```diff
@@ -129,3 +129,32 @@ jobs:
...
+        - name: Download Build Artifacts
+          uses: actions/download-artifact@v1
```
Comment (sfc-gh-pbulawa) [inline .github/workflows/build-test.yml:144]: Could you use v3 version of download-artifact?

File: ci/test_ubuntu.sh (hunk 3 above)
Comment (sfc-gh-pmotacki) [reply ci/test_ubuntu.sh:11]: Yes we should.

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (sfc-gh-dprzybysz) [reply .github/workflows/build-test.yml:151]: For mac we are not doing it. Additionally I see that test_component.sh install the dependencies so this step seems to be unnecessary

File: ci/test_mac.sh [hunk 6]
```diff
@@ -4,7 +4,6 @@
 #
 set -o pipefail
 THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-source $THIS_DIR/_init.sh
```
Comment (sfc-gh-dprzybysz) [inline ci/test_mac.sh:7]: i don't think we should change mac build here

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (sfc-gh-pmotacki) [reply .github/workflows/build-test.yml:151]: ok

File: .github/workflows/build-test.yml (hunk 4 above)
Comment (sfc-gh-pmotacki) [reply .github/workflows/build-test.yml:142]: yes

File: ci/test_mac.sh (hunk 6 above)
Comment (sfc-gh-pmotacki) [reply ci/test_mac.sh:7]: ok

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Pull request merged**